presto-main/src/main/java/com/facebook/presto/operator/FileFragmentResultCacheManager.java

```diff
@@ -146,13 +152,18 @@ private static long getPagesSize(List<Page> pages)
                 .sum();
     }
 
-    private void cachePages(CacheKey key, Path path, List<Page> pages)
+    private void cachePages(CacheKey key, Path path, List<Page> pages, long resultSize)
     {
         // TODO: To support both memory and disk limit, we should check cache size before putting to cache and use written bytes as weight for cache
         try {
             Files.createFile(path);
             try (SliceOutput output = new OutputStreamSliceOutput(newOutputStream(path, APPEND))) {
-                writePages(pagesSerde, output, pages.iterator());
+                if (isBlockEncodingCompressionEnabled) {
+                    writePages(pagesSerdeFactory.createPagesSerde(), output, pages.iterator());
```

Comment:
Curious: Does it make sense that instead of creating `PagesSerde` every time, we create only once in the constructor, and directly replace current `pagesSerde`

Reply:
Alternatively, you could use something similar to the changes I made in Trino https://github.com/trinodb/trino/pull/5545 which (mostly) puts the not thread safe buffers into a context object that can be created / released. I say "mostly" here because I didn't move the `Compressor` instance in which you would need to do in order to use it that way.

Another thing worth pointing out is that this implementation, as written, doesn't support encryption of the serialized pages.

Reply:
@shixuan-fan i have delete the write compress buffer and relation code.I find it not work;

@pettyjamesm thanks for you advice .

Reply:
> @shixuan-fan i have delete the write compress buffer and relation code.I find it not work.Can you help me approve

I'm trying to understand this. Do you mean you tried @pettyjamesm 's suggestion but it does not work?